src/bin/agent/CommandLockAll.ts

```diff
@@ -39,27 +42,37 @@ class CommandLockAll extends CommandPolykey {
         fs: this.fs,
         logger: this.logger.getChild(Session.name),
       });
-      let pkClient: PolykeyClient;
+      let webSocketClient: WebSocketClient;
+      let pkClient: PolykeyClient<typeof clientManifest>;
       this.exitHandlers.handlers.push(async () => {
         if (pkClient != null) await pkClient.stop();
+        if (webSocketClient != null) await webSocketClient.destroy(true);
       });
       try {
-        pkClient = await PolykeyClient.createPolykeyClient({
-          nodeId: clientOptions.nodeId,
+        webSocketClient = await WebSocketClient.createWebSocketClient({
+          expectedNodeIds: [clientOptions.nodeId],
           host: clientOptions.clientHost,
           port: clientOptions.clientPort,
+          logger: this.logger.getChild(WebSocketClient.name),
+        });
+        pkClient = await PolykeyClient.createPolykeyClient({
+          streamFactory: () => webSocketClient.startConnection(),
           nodePath: options.nodePath,
+          manifest: clientManifest,
```

Comment:
There should be a way of acquiring this information without importing the entire application. I mean also in a transitive sense.

Reply:
For the bin commands we can just define the types for the RPC methods we're using in the command. The problem is, it's essential that they match otherwise very annoying to debug bugs happen. To mitigate this the caller and handler are defined side by side in the same location. But as you said this leaves us depending on a bunch of stuff we don't need. 

We can just define the caller in the command and use a truncated manifest, Or separate the client manifest definition to a new file so it only depends on the types it needs.

Reply:
This is the same problem as the above comment. I'll do the discussion there.